### PR TITLE
Game chat overhaul

### PR DESCRIPTION
**Pull request by uzher-code**
Made the chat User Interface more user-friendly. Changed states being used. Added socket room for game chat in the backend socket.js so that each game has its own room/channel, and players do not see chat from a different game. Made sure that scroll ability is activated once the chat gets to the end of the chat window

Added a different color for the messages from the current logged in use:

Light Salmon color is for the user that is current user
Light Slate Grey is for all other players

image

### CODE REVIEW

File: client/src/components/Chat.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    },
+});
+
+const Chat = () => {
+    const [message, setMessage] = useState('');
+	const [ chat, setChat ] = useState([]);
+    const { user } = useUserState();
+    const { matchId } = useParams();
+	const socketRef = useRef();
+    const classes = useStyles();
```
Comment (dproc96) [inline client/src/components/Chat.js]: Can you run prettier and update the spacing here?

File: client/src/components/Chat.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+              <Box pt={2}>
+                <TextField
+                  className={classes.message}
+                  name="message"
+                  onChange={(e) => onTextChange(e)}
+                  value={message}
+                  id="outlined-multiline-static"
+                  variant="outlined"
+                  label="Message"
+                  onKeyDown={(event) => handleKeyDown(event)}
```
Comment (dproc96) [inline client/src/components/Chat.js]:
```suggestion
                  onKeyDown={handleKeyDown}
```
Ditto this for the other event functions. You can just pass the function as a callback

File: server/socket.js [hunk 3]
```diff
@@ -5,9 +5,11 @@ var socket = require("socket.io");
...
     const io = socket(server);
     io.on("connection", socket => {
-        socket.on("message", ({ name, message }) => {
-            io.emit("message", { name, message });
-            console.log(name + " has connected.");
+        socket.on('join-chat', ({ matchId }) => {
+         socket.join(matchId);
+        });
+        socket.on("message", ({ name, message, matchId }) => {
+            io.to(matchId).emit('message', { name, message });
```
Comment (dproc96) [inline server/socket.js:12]: Nice use of socket rooms 👍

**Review by dproc96 — COMMENTED** [review]
Can you run prettier on these files? It looks like there's some spacing that's off

File: client/src/components/Chat.js (hunk 1 above)
Comment (uzher-code) [reply client/src/components/Chat.js]: used prettier to fix spacing

File: client/src/components/Chat.js (hunk 2 above)
Comment (uzher-code) [reply client/src/components/Chat.js]: made use of callback now

**Review by dproc96 — APPROVED** [review]
This looks good to me!

**Pull request merged**